google/cloud/aiplatform/training_jobs.py

```diff
@@ -1185,4 +1177,207 @@ def _model_upload_fail_string(self) -> str:
+
+                Regression:
+                "minimize-rmse" (default) - Minimize root-mean-squared error (RMSE).
+                "minimize-mae" - Minimize mean-absolute error (MAE).
+                "minimize-rmsle" - Minimize root-mean-squared log error (RMSLE).
+            column_transformations (Optional[Union[Dict, List[Dict]]]):
+                Optional. Transformations to apply to the input columns (i.e. columns other
+                than the targetColumn). Each transformation may produce multiple
+                result values from the column's value, and all are used for training.
+                When creating transformation for BigQuery Struct column, the column
+                should be flattened using "." as the delimiter.
+                If an input column has no transformations on it, such a column is
+                ignored by the training, except for the targetColumn, which should have
+                no transformations defined on.
+            optimization_objective_recall_value (float):
+                Optional. Required when maximize-precision-at-recall optimizationObjective was
+                picked, represents the recall value at which the optimization is done.
+
+                The minimum value is 0 and the maximum is 1.0.
+            optimization_objective_precision_value (float):
+                Optional. Required when maximize-recall-at-precision optimizationObjective was
+                picked, represents the precision value at which the optimization is
+                done.
+
+                The minimum value is 0 and the maximum is 1.0.
+            project (str):
+                Optional. Project to run training in. Overrides project set in aiplatform.init.
+            location (str):
+                Optional. Location to run training in. Overrides location set in aiplatform.init.
+            credentials (auth_credentials.Credentials):
+                Optional. Custom credentials to use to run call training service. Overrides
+                credentials set in aiplatform.init.
+        """
+        super().__init__(
+            display_name=display_name,
+            project=project,
+            location=location,
+            credentials=credentials,
+        )
+        self._column_transformations = column_transformations
+        self._optimization_objective = optimization_objective
+        self._optimization_prediction_type = optimization_prediction_type
+        self._optimization_objective_recall_value = optimization_objective_recall_value
+        self._optimization_objective_precision_value = (
+            optimization_objective_precision_value
+        )
+
+    def run(
+        self,
+        dataset: Optional[datasets.Dataset],
+        target_column: str,
+        training_fraction_split: float,
+        validation_fraction_split: float,
+        test_fraction_split: float,
+        weight_column: Optional[str] = None,
+        budget_milli_node_hours: int = 1000,
```

Comment:
I'm guessing setting this to less than 1000 hours will cause the backend to use 1000 hours instead. Probably same with the maximum value.

I have no idea if this is true though and this seems opaque to users.

Ideally, the backend should respond with a warning if the parameters are out of bounds.